### DISCUSSION TITLE
[SOLVED] +clone not working as before since 7.0.11.7?

### DISCUSSION THREAD

**Discussion by abique** [ImageMagick/ImageMagick#3561] [Help]
Hi,

The following command used to work:
```
'convert' '-background' 'transparent' '-density' '90' 'touch.svg' '(' '+clone' '-background' 'black' '-shadow' '50x1+1+1' ')' '-compose' 'DstOver' '-background' 'none' '-layers' 'flatten' '+repage' '-quality' '100%' 'touch.png'
```
But now it fails:
```
convert: invalid image index `touch.svg' @ error/list.c/CloneImages/282.
convert: no such image `+clone' @ error/convert.c/ConvertImageCommand/1027.
```

I'm not a `convert` expert, and I don't understand what that command was doing. Could someone tell me if it is a regression, and if not, how to change the command to make it work again.

Thank you very much.

**Comment by fmw42** [comment]
On Imagemagick 7, use magick not convert.  What is your exact IM version? Why do you have all those quotes? Which SVG render are you using:   MSVG, RSVG, Inkscape?  I believe the error is with reading the SVG file not +clone.

**Reply by abique** [reply]
I've tried a bunch of things and I replaced `+clone` by `-clone 0`. It seems to work.
I have quotes because this command is printed by a script.
I can call magick instead, does it change something?
For the svg render, I don't know.

Thanks!

**Reply by fmw42** [reply]
magick is the new convert for IM 7.  Using convert will use IM 6 syntax. In your case it might not make a difference.  Post your SVG file so the IM developers can test with it. You may need to zip it first.

**Reply by Jmuccigr** [reply]
I am experiencing the same issue with a long-running script that now barfs on `+clone`, but works with `-clone 0`. The exact error is:

```
magick: invalid image index `filename.png' @ error/list.c/CloneImages/282.
magick: no such image `+clone' at CLI arg 5 @ error/operation.c/CLINoImageOperator/5129.
```

`identify format "%n" filename.png` yields 1, as expected.

MacOS 10.15.7
Version: ImageMagick 7.0.11-7 Q16 x86_64 2021-04-12 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib

**Reply by fmw42** [reply]
What was the command that failed with +clone

**Reply by fmw42** [reply]
This seems to work fine for me on IM 7.0.11.8 Q16 Mac OSX Sierra:

` magick rose: \( +clone  -background navy  -shadow 80x3+5+5 \) +swap -background none   -layers merge  +repage   shadow.png
`

What is your platform?

**Reply by Jmuccigr** [reply]
> What was the command that failed with +clone

I was getting the error on a simplified version of the actual command: `magick filename.png \( +clone -negate \) -compose divide_src -composite  +repage -white-threshold 90% output.png`

I'm using homebrew and there was a new version of IM today which I upgraded to. Right now your simple clone with rose: works for me. I don't have the original of the file that wasn't working, but I did recreate it and it works fine too. So it looks like something got fixed.

```
Version: ImageMagick 7.0.11-8 Q16 x86_64 2021-04-17 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
```
